src/http-blockstore.ts

```diff
@@ -0,0 +1,109 @@
+import type { Blockstore, Pair } from 'interface-blockstore'
+import type { Connection } from '@libp2p/interface-connection'
+import type {
+  AbortOptions,
+  Await,
+  AwaitIterable
+} from 'interface-store'
+import type { Libp2p } from 'libp2p'
+import type { CID } from 'multiformats/cid'
+import { fetchViaDuplex } from '@marcopolo_/libp2p-fetch'
+
+// TODOS:
+// - Validate the blocks we fetch
+// - Use format=car to get the whole dag at once rather than per block (probably faster?)
+// - Keep a list of HTTPS endpoints, not just libp2p peers.
+// - Keep a list of previous peers we were connected to, so even if we lose a webtransport connection, we can spin it up later.
+
+export class HttpBlockstore implements Blockstore {
```

Comment:
IIUC this approach will cause problems when also enabling fetching over Bitswap (I think it might check the blockstore before doing a Bitswap request which would basically cause these lookups to be sequential).

An alternative way to do this would be (although @achingbrain might have more helia-friendly suggestions):
1. Implement this as the Bitswap interface (i.e. want, unwant, cancel_wants)
2. Make a third implementation of the Bitswap interface that basically asks both in parallel, but when one of them yields a block it cancels on the other one

Later we can basically do combined task prioritization with Bitswap whereby we hit peers (Bitswap or HTTP or HTTP over libp2p) based on how quickly they respond to us with good data.